src/commands/handlers/misc.js

```diff
@@ -100,52 +93,73 @@ const handlers = {
             hostname: params[3],
             server: params[4],
             real_name: realname,
-            away: is_away,
             num_hops_away: hops_away,
             channel: params[1],
-            channel_modes: chan_modes,
-            tags: command.tags
+            tags: command.tags,
+            unparsed_flags: unparsedFlags,
+            ...parsedFlags,
         });
     },
 
     RPL_WHOSPCRPL: function(command, handler) {
         const cache = handler.cache('who');
         if (!cache.members) {
             cache.members = [];
+            cache.type = 'whox';
         }
+
         const params = command.params;
 
-        // G = Gone, H = Here
-        const is_away = params[6][0].toUpperCase() === 'G';
+        if (cache.token === 0) {
+            // Token validation has already been attempted but failed,
+            // ignore this event as it will not be emitted
+            return;
+        }
+
+        if (!cache.token) {
+            const token = parseInt(params[1], 10) || 0;
```

Comment:
You should probably also check if the WHOX response has enough parameters just in case someone accidentally reuses your token.